marklogic-client-api/src/main/java/com/marklogic/client/impl/OkHttpServices.java

```diff
@@ -247,13 +248,7 @@ public void connect(String host, int port, String database, SecurityContext secu
     this.configureHostnameVerifier(clientBuilder, sslVerifier);
 
     this.database = database;
-
-    this.baseUri = new HttpUrl.Builder()
-      .scheme(sslContext == null ? "http" : "https")
-      .host(host)
-      .port(port)
-      .encodedPath("/v1/ping")
-      .build();
+    this.baseUri = HttpUrlBuilder.newBaseUrl(host, port, basePath, sslContext);
```

Comment:
This is the main change here, along with what's below - all the stuff in HttpUrlBuilder is what really matters.